src/servers/src/grpc/greptime_handler.rs

```diff
@@ -161,11 +162,13 @@ pub(crate) fn create_query_context(header: Option<&RequestHeader>) -> QueryConte
             }
         })
         .unwrap_or((DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME));
-
-    QueryContextBuilder::default()
+    let timezone = parse_timezone(header.map(|h| h.timezone.as_str()));
+    let ctx = QueryContextBuilder::default()
         .current_catalog(catalog.to_string())
         .current_schema(schema.to_string())
-        .build()
+        .build();
+    ctx.set_timezone(timezone);
```

Comment:
What about adding a builder function `timezone(timezone)` into `QueryContextBuilder`? If so, we don't need `ctx.set_timezone` any more. The code style would be more consistent.